src/codegen/translate-expression.lisp

```diff
@@ -34,9 +39,7 @@ TRANSLATE-EXPRESSION when an abstraction is being translated.")
            (type symbol name)
            (type pred-context extra-context))
 
-  (let* ((*current-function-name* name)
```

Comment:
this should only be bound when the `name` represents a `node-abstraction` (first two cases of the `cond` here).